Libs/MRML/Logic/vtkMRMLColorLogic.cxx

```diff
@@ -1076,3 +1094,699 @@ vtkMRMLProceduralColorNode* vtkMRMLColorLogic::CopyProceduralNode(vtkMRMLColorNo
+    return false;
+    }
+  colorNode->SetAttribute("TerminologyName", lutName.c_str());
+
+  return true;
+}
+//------------------------------------------------------------------------------
+bool vtkMRMLColorLogic::InitializeTerminologyMappingFromFile(std::string mapFileName)
+{
+  std::cout << "Initializing terminology mapping for map file " << mapFileName << std::endl;
+
+  std::ifstream mapFile(mapFileName.c_str());
+  bool status = mapFile.is_open();
+  std::string lutName = "";
+  bool addFlag = true;
+  bool assocFlag = true;
+  if (status)
+    {
+    while (!mapFile.eof())
+      {
+      std::string lineIn;
+      std::getline(mapFile, lineIn);
+
+      if (lineIn[0] == '#')
+        {
+        continue;
+        }
+      if (lineIn.find("SlicerLUT=") == std::string::npos)
+        {
+        continue;
+        }
+      size_t delim = lineIn.find("=");
+      lutName = lineIn.substr(delim+1,lineIn.length()-delim);
+      assocFlag = this->CreateNewTerminology(lutName);
+      break;
+      }
+
+    while (!mapFile.eof())
+      {
+      std::string lineIn, lineLeft;
+      std::getline(mapFile, lineIn);
+      if (lineIn.length()<30 || lineIn[0] == '#')
+        {
+        continue;
+        }
+      std::vector<std::string> tokens;
+      std::stringstream ss(lineIn);
+      std::string item;
+      while (std::getline(ss,item,','))
+        {
+        tokens.push_back(item);
+        }
+
+      int labelValue = atoi(tokens[0].c_str());
+      StandardTerm category, type, modifier;
+      this->ParseTerm(tokens[2],category);
+      this->ParseTerm(tokens[3],type);
+      this->ParseTerm(tokens[4],modifier);
+
+      // for now region doesn't appear in the file
+      StandardTerm region, regionModifier;
+      addFlag = addFlag && this->AddTermToTerminologyMapping(lutName, labelValue, category, type, modifier, region, regionModifier);
+     }
+  }
+  std::cout << this->colorCategorizationMaps[lutName].size()
+            << " terms were read for Slicer LUT " << lutName << std::endl;
+
+  return status && addFlag && assocFlag;
+}
+
+//-------------------------------------------------------------------------------
+bool vtkMRMLColorLogic::
+  LookupCategorizationFromLabel(int label, ColorLabelCategorization& labelCat, const char *lutName)
+{
+  bool success = false;
+
+
+  if (this->TerminologyExists(lutName))
+    {
+    // set the label value so that if it's not found, it's still a valid categorisation
+    labelCat.LabelValue = label;
+    if (this->colorCategorizationMaps[lutName].find(label) !=
+      this->colorCategorizationMaps[lutName].end())
+      {
+      labelCat = this->colorCategorizationMaps[lutName][label];
+      success = true;
+      }
+    }
+  return success;
+}
+
+//---------------------------------------------------------------------------
+std::string vtkMRMLColorLogic::
+  GetTerminologyFromLabel(std::string categorization, std::string standardTerm, int label, const char *lutName)
```

Comment:
Along the comments JC did below, it makes sense to use enums instead of strings here.